### PR TITLE
GithubService:: `commentResults` and `getMatchedLinesAsLink` tests

### PR DESCRIPTION
**Pull request by namefreezers**
# Description

Added unit test set for the com.lpvs.service.GithubService class. This step is required for the GitHub action infra setup.

## Type of change

Please delete options that are not relevant.

- [x] Code cleanup/refactoring
- [x] Test Coverage update

# How Has This Been Tested?
mvn clean install
mvn test

**Test Configuration**:
* Java: v18
* LPVS Release: v1.0.0

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by o-kopysov — APPROVED** [review]
Great job!
Current coverage is 53.2%

File: src/test/java/com/lpvs/service/GitHubServiceTest.java [hunk 1]
```diff
@@ -1079,6 +1123,460 @@ public void testSetErrorCheck__ApiUrlPresentCantAuthorize() {
...
+                " * found in the LICENSE file.\n" +
+                " */\n";
+        final String matched_lines_1 = "1-6";
+        final String component_1 = "LPVS::Services";
+
+        // `lpvs_file_1`
+        LPVSLicense lpvs_license_1;
+        final String license_name_1 = "MIT License";
+        final String spdx_id_1 = "MIT";
+        final String access_1 = "PROHIBITED";
```
Comment (MoonkiHong) [inline src/test/java/com/lpvs/service/GitHubServiceTest.java:1362]: Do we need to explosure the policy (in this code `access`) in public?

**Review by MoonkiHong — COMMENTED** [review]
Please take a look at my comment. 🙏

File: src/test/java/com/lpvs/service/GitHubServiceTest.java (hunk 1 above)
Comment (o-kopysov) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:1362]: To cover the code related to the processing of the detected prohibited license and the license conflict, we are simulating such a case in this unit test.
This case is an artificial and has no relation to the real policy.

Comment (namefreezers) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:1362]: Oleg is right. These are just random values for unit test.
I can change them to other random values, should I do this?

Comment (MoonkiHong) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:1362]: Then the presented `access_1` can be anything, right? What about `CASE1` or `CASE2` or something in this case?

Comment (o-kopysov) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:1362]: The value of `access_1` could be one among the list: "PERMITTED", "RESTRICTED", or "PROHIBITED".

Comment (namefreezers) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:1362]: https://github.com/Samsung/LPVS/blob/443ed9ddeeda83d0da09c6daf878350d33863bf1/src/main/java/com/lpvs/service/GitHubService.java#L176

we need it to get into this `if` branch

Comment (MoonkiHong) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:1362]: Is the current implementation dependent on the values in `access` then? Please educate me if this test is in the end for our current implementation in conjunction with our previous discussion in https://github.com/Samsung/LPVS/pull/47. 🙏

Comment (o-kopysov) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:1362]: Conclusion of the LPVS about the usage of prohibited or restricted licenses depends on the file [licenses.json](https://github.com/Samsung/LPVS/blob/main/src/main/resources/licenses.json)
According to the README file, a user has to fill this file by himself before starting to use LPVS. We provide only a template for this file.
Here in unit test we simulate values from this `licenses.json` file using variables `license_name_1`, `spdx_id_1`, `access_1`, etc., which is equivalent that we read them from the file.

Comment (MoonkiHong) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:1362]: +1 Thank you for your kind clarification. 😄

**Review by MoonkiHong — APPROVED** [review]
+1

**Pull request merged**